core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -778,6 +786,13 @@ public InodeDirectory getRoot() {
     }
     newInode.setPinned(currentInodeDirectory.isPinned());
 
+    if (ServerConfiguration.getBoolean(PropertyKey.MASTER_METASTORE_INODE_INHERIT_OWNER_AND_GROUP)
```

Comment:
Does it matter that this is performed after the acls, but the previous edit is done before the acls?

Reply:
i don't think so. @yuzhu can you confirm please?

Reply:
in both usages, the empty check is after setOwner(pathContext.getOwner)

Reply:
If we want the ACL inheritance to override the `MASTER_METASTORE_INODE_INHERIT_OWNER_AND_GROUP` property (I think we should), we need to have the ACL block after this section of the code. Otherwise, whatever ACL sets will be overwritten by the call to `setOwner` and `setGroup` calls.

Reply:
thx david